lua/controller.lua

```diff
@@ -253,49 +258,68 @@ local function controller_get_drawer_index(pos, itemstring)
 	return drawer_net_index
 end
 
+local function controller_insert_to_empty_drawer(pos, stack)
+	local name = stack:get_name()
+	local drawer_net_index = controller_get_drawer_index(pos, name)
+
+	if not drawer_net_index["empty"] then return stack end
+
+	local drawer_pos = drawer_net_index["empty"]["drawer_pos"]
+	local visualid = drawer_net_index["empty"]["visualid"]
+	local content = drawers.drawer_get_content(drawer_pos, visualid)
+
+	-- If the drawer is still empty and the drawer entity is loaded, we will
+	-- put the items in the drawer
+	if content.name == "" and drawers.drawer_visuals[core.serialize(drawer_pos)] then
+		local leftover = drawers.drawer_insert_object(drawer_pos, stack, visualid)
+
+		-- Add the item to the drawers table index and set the empty one to nil
+		drawer_net_index["empty"] = nil
+		drawer_net_index[name] = controller_index_slot(drawer_pos, visualid)
+
+		-- Set the controller metadata
+		core.get_meta(pos):set_string("drawers_table_index",
+									  core.serialize(drawer_net_index))
```

Comment:
because the line becomes too long and thus hard to read

Reply:
There are many other lines that are longer than this, and luacheck doesn't complain about them...

Reply:
Lua can handle a lot longer lines than fit in editor window width.

Reply:
In that case you also need to change lines 69, 164, 169, 357, and 373; they are all longer that this would be. Do you see what my point is?

Reply:
> In that case you also need to change lines 66, 69, 164, 169, 357, 358, and 373; they are all longer that this would be. Do you see what my point is?